examples/run-blockchain/index.ts

```diff
@@ -0,0 +1,118 @@
+import VM from '../../'
+
+import Account from 'ethereumjs-account'
+import * as utils from 'ethereumjs-util'
+import { promisify } from 'util'
+
+const Trie = require('merkle-patricia-tree/secure')
+const Block = require('ethereumjs-block')
+const Blockchain = require('ethereumjs-blockchain')
+const BlockHeader = require('ethereumjs-block/header.js')
+const level = require('level')
+const levelMem = require('level-mem')
+const testData = require('./test-data')
+
+const rlp = utils.rlp
+
+async function main() {
+  const state = new Trie()
+  const blockchainDB = levelMem()
+
+  const hardfork = testData.network.toLowerCase()
+
+  const blockchain = new Blockchain({ db: blockchainDB, hardfork })
+  blockchain.ethash.cacheDB = level('./.cachedb')
+
+  const vm = new VM({
+    state: state,
+    blockchain: blockchain,
+    hardfork,
+  })
+
+  vm.on('beforeTx', (tx: any) => {
+    tx._homestead = true
+  })
+
+  vm.on('beforeBlock', (block: any) => {
+    block.header.isHomestead = function() {
+      return true
+    }
+  })
+
+  await setupPreConditions(state, testData)
+
+  await setGenesisBlock(blockchain, hardfork)
+
+  await putBlocks(blockchain, hardfork, testData)
+
+  await vm.runBlockchain(blockchain)
+
+  const blockchainHead = await promisify(vm.blockchain.getHead.bind(vm.blockchain))()
+  state.root = blockchainHead.header.stateRoot
+
+  console.log('--- Finished processing the BlockChain ---')
+  console.log('New head:', '0x' + blockchain.meta.rawHead.toString('hex'))
+  console.log('Expected:', testData.lastblockhash)
+}
+
+async function setupPreConditions(state: any, testData: any) {
+  for (const address of Object.keys(testData.pre)) {
+    const acctData = testData.pre[address]
+    const account = new Account()
+
+    account.nonce = utils.toBuffer(acctData.nonce)
+    account.balance = utils.toBuffer(acctData.balance)
+
+    const storageTrie = state.copy()
+    storageTrie.root = null
+
+    for (const hexStorageKey of Object.keys(acctData.storage)) {
+      const val = utils.toBuffer(acctData.storage[hexStorageKey])
+      const storageKey = utils.setLength(utils.toBuffer(hexStorageKey), 32)
+
+      await promisify(storageTrie.put.bind(storageTrie))(storageKey, rlp.encode(val))
+    }
+
+    const codeBuf = utils.toBuffer(acctData.code)
+
+    await setCode(account, state, codeBuf)
+
+    account.stateRoot = storageTrie.root
```

Comment:
Is setting the root enough? How so?

Reply:
Let me know if this wasn't clear after my previous comment